### PR TITLE
Tyler Blackmon (completed IC#3)

### PR DESCRIPTION
**Pull request by tblackmon-tiel**
Functionality for iteration 4 is in the PatronService class

### CODE REVIEW

**Comment by tblackmon-tiel** [comment]
I went back and added the iteration 4 methods to individual parks as well since I had time at the end, wasn't sure which way it wanted!

File: lib/passenger.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Passenger
+  attr_reader :name, :age
+  def initialize(details)
+    @name = details["name"]
+    @age = details["age"]
+    @driver = false
+  end
+
+  def adult?
+    @age >= 18 ? true : false
```
Comment (dcoleman21) [inline lib/passenger.rb:10]: You don't even need to use the ternary here. The `>=` operator will return either true or false.

File: lib/vehicle.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  def speed
+    @speeding = true
+  end
+
+  def add_passenger(passenger)
+    @passengers << passenger
+  end
+
+  def num_adults
+    @passengers.count { |passenger| passenger.adult? }
```
Comment (dcoleman21) [inline lib/vehicle.rb:28]: Nice job with the `.count` enum

File: spec/park_spec.rb [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    end
+  end
+
+  describe "#add_vehicle" do
+    it "adds a vehicle to the park" do
+      park = Park.new("Mesa Verde", 10)
+      fusion = Vehicle.new("2020", "Ford", "Fusion")
+      yukon = Vehicle.new("2005", "GMC", "Yukon")
+
+      expect(park.vehicles).to be_empty
```
Comment (dcoleman21) [inline spec/park_spec.rb:35]: Nice assertion here

File: spec/park_spec.rb [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+      yukon.add_passenger(bill)
+      yukon.add_passenger(josh)
+
+      park.add_vehicle(fusion)
+      park.add_vehicle(yukon)
+
+      expect(park.adults).to eq(["Bill", "Josh", "Tyler"])
+    end
+  end
+end
```
Comment (dcoleman21) [inline spec/park_spec.rb:173]: All these tests look great.

**Pull request closed without merging**